File: lib/antlr-2.7.5/examples/csharp/imagNodeAST/Main.cs
namespace ImagNodeTest
{
	using System;
	using System.IO;
	using antlr;
	
	class AppMain {
		public static void Main(string[] args) {
			try {
				LangLexer lexer = new LangLexer(new CharBuffer(Console.In));
				LangParser parser = new LangParser(lexer);
				parser.block();
				CommonAST a = (CommonAST) parser.getAST();
				Console.Out.WriteLine(a.ToStringList());
				LangWalker walker = new LangWalker();
				walker.block(a);	// walk tree
				Console.Out.WriteLine("done walking");
			} catch(Exception e) {
				Console.Error.WriteLine("exception: "+e);
			}
		}
	}
}

File: lib/antlr-2.7.5/examples/java/imagNodeAST/Main.java
import java.io.*;
import antlr.DumpASTVisitor;

class Main {
	public static void main(String[] args) {
		try {
			LangLexer lexer = new LangLexer(new DataInputStream(System.in));
			LangParser parser = new LangParser(lexer);
			parser.block();
			antlr.CommonAST a = (antlr.CommonAST)parser.getAST();
			System.out.println(a.toStringList());
			LangWalker walker = new LangWalker();
			walker.block(a);	// walk tree
			System.out.println("done walking");
		} catch(Exception e) {
			System.err.println("exception: "+e);
		}
	}
}

